core/src/main/java/org/infinispan/util/concurrent/BlockingTaskAwareExecutorServiceImpl.java

```diff
@@ -29,13 +29,16 @@
    private final BlockingQueue<BlockingRunnable> blockedTasks;
    private final ExecutorService executorService;
    private final TimeService timeService;
+   private final ControllerThread controllerThread;
    private volatile boolean shutdown;
 
-   public BlockingTaskAwareExecutorServiceImpl(ExecutorService executorService, TimeService timeService) {
-      this.blockedTasks = new LinkedBlockingQueue<BlockingRunnable>();
+   public BlockingTaskAwareExecutorServiceImpl(String controllerThreadName, ExecutorService executorService, TimeService timeService) {
+      this.blockedTasks = new LinkedBlockingQueue<>();
       this.executorService = executorService;
       this.timeService = timeService;
       this.shutdown = false;
+      this.controllerThread = new ControllerThread(controllerThreadName);
+      controllerThread.start();
```

Comment:
Well, WildFly 9 has deprecated the thread subsystem and all subsystems requiring their own threads/executors need to handle them themselves, so I think that is not a hurdle.